flocker/provision/_digitalocean.py

```diff
@@ -0,0 +1,224 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+DigitalOcean provisioner.
+"""
+import time
+from functools import partial
+
+import pyocean
+
+from ._libcloud import LibcloudProvisioner
+from ._install import (
+    provision, run,
+    task_install_kernel,
+)
+
+
+def retry_if_pending(callable, *args, **kwargs):
+    """
```

Comment:
:question: You might have an easier time if you use `flocker.testtools.loop_until`.